Excercise 0.5.md

```diff
@@ -0,0 +1,47 @@
+sequenceDiagram
+    participant browser
+    participant server
+
+    browser->>server:Executes the JavaScript code fetched from the server
+    activate server
+    server-->>browser:fetches the notes as JSON data and adds HTML elements for displaying the notes
+    deactivate server
+
+    Note left of server:uses DOM-API
+
+    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/notes
+    activate server
+    server-->>browser: the contents of are manipulated with JavaScript that executes in the browser.
+    deactivate server
+
+    Note right of browser:comprise only one HTML page
+
+    browser->>server:data is sent via form's submit
+    activate server
+    server-->>browser:instructs to reload the Notes page with a redirect
+    deactivate server
+
+    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/spa
```

Comment:
@ArpitAgrawal001 This is where the single page app flow actually begins. Everything before this line is the traditional flow.